tests/Type/Doctrine/Collection/data/collection.php

```diff
@@ -26,6 +26,13 @@ class MyEntity
 
 	$false2 = $collection->last();
 	$false2;
+
+	$collection->add($new);
+	$result1 = $collection->first();
+	$result1;
+
+	$result2 = $collection->last();
+	$result2;
 }
```

Comment:
Please do not modify the test like that, the diff of the rule test case is unreadable. Also, it'd make sense to refactor this into a TypeInferenceTestCase :) Thank you.

Reply:
The test is now refactored into a TypeInferenceTestCase ;)